python/lsst/ts/m2/csc.py

```diff
@@ -122,6 +123,13 @@ def __init__(
         simulation_mode: int = 0,
         verbose: bool = False,
     ) -> None:
+        # This is to keep the backward compatibility of ts_xml v22.0.0 that
+        # does not have the 'enableLutTemperature' command defined in xml.
+        # TODO: Remove this after ts_xml v22.1.0.
+        component_info = ComponentInfo("MTM2", "sal")
+        if "cmd_enableLutTemperature" in component_info.topics:
```

Comment:
I don't like this coding style except in cases it's really needed. Do you really need to check if the command is there? It's problem of the deployment policy to not deploy apps not matching XML schema. The default behavior if the command isn't in topic would be an error, which tells you more - either roll back old application, or wait for XML update. This just add complexity to the code when it's not needed.

Reply:
I do not like this as well and had suggested Tiago to have a better way to deal with the backward compatibility in the future's Kafka version. Before the `ts_xml` and `ts_salobj` have the further support, we might need to live with this workaround for a while. Thanks!